### PR TITLE
ARROW-15472: [Website] Add Flight SQL announcement

### PR DESCRIPTION
**Pull request by lidavidm**

### CODE REVIEW

**Comment by lidavidm** [comment]
<details>
<summary>Rendered</summary>

image

</details>

File: _posts/2022-02-04-introducing-arrow-flight-sql.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+---
+layout: post
+title: "Introducing Apache Arrow Flight SQL: Accelerating Database Access"
+description: "This post introduces Arrow Flight SQL, a protocol for interacting
+with SQL databases over Arrow Flight. We have been working on this protocol
+over the last six months, and are looking for feedback, interested
+contributors, and early adopters."
+date: "2022-02-04 00:00:00 -0500"
+author: James Duong, Kyle Porter, David Li
```
Comment (lidavidm) [inline _posts/2022-02-04-introducing-arrow-flight-sql.md:9]: This list needs to be extended.

File: _posts/2022-02-04-introducing-arrow-flight-sql.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+Note that while Flight SQL is shipping as part of Apache Arrow 7.0.0, it is still under development, and detailed documentation is forthcoming.
+However, implementations are already available in C++ and Java, which provide a low-level client that can be used as well as a server skeleton that can be implemented.
+
+For those interested, a [server implementation wrapping Apache Derby](https://github.com/apache/arrow/blob/release-7.0.0/java/flight/flight-sql/src/test/java/org/apache/arrow/flight/sql/example/FlightSqlExample.java) and [one wrapping SQLite](https://github.com/apache/arrow/blob/release-7.0.0/cpp/src/arrow/flight/sql/example/sqlite_server.h) are available in the source.
+A [simple CLI demonstrating the client](https://github.com/apache/arrow/blob/release-7.0.0/cpp/src/arrow/flight/sql/test_app_cli.cc) is also available.
+
+## What's Next & Getting Involved
+
+Compared to existing libraries like PyODBC, [Arrow Flight is already as much as 20x faster][subsurface] (~00:21:00).
+Flight SQL will package these performance advantages into a standard interface, ready for clients and databases to implement.
```
Comment (lidavidm) [inline _posts/2022-02-04-introducing-arrow-flight-sql.md:154]: Is there a better or other examples of the benefits of Flight SQL?

**Comment by lidavidm** [comment]
CC @kylepbit @jduo @rafael-telles @jcralmeida @vfraga (and anyone else I may have missed), I've drafted a post about Flight SQL and would appreciate any feedback/contributions. (You can leave it here or submit a PR against https://github.com/lidavidm/arrow-site/tree/arrow-15472)

**Comment by wesm** [comment]
I gave this a read through and think it does a good job presenting the subject. It's admittedly a nuanced topic because Flight SQL is not _exactly_ a replacement for ODBC/JDBC since those are APIs are not wire protocols / driver implementations, so maybe creating a diagram to show that FlightSQL is occupying the "middle layer" of a database driver implementation (the internal pieces that stand between the ODBC/JDBC API façade and the server/client). Of course if you wanted FlightSQL to be your database's client API, you are certainly free to do that.

**Comment by lidavidm** [comment]
Revised, re-rendered:

<details>

image

</details>

**Comment by lidavidm** [comment]
CC @jduo @kylepbit if either of you have comments or would like to contribute

File: _posts/2022-02-04-introducing-arrow-flight-sql.md (hunk 1 above)
Comment (rafael-telles) [reply _posts/2022-02-04-introducing-arrow-flight-sql.md:9]: Please add José Almeida, Juscelino Junior, Rafael Telles and Vinicius Fraga

**Comment by rafael-telles** [comment]
Very nice post @lidavidm ! Just a minor comment: the code example on "Getting Started" prints to the console "Read one **chunk**" but ends up printing the whole table.

**Comment by lidavidm** [comment]
Thanks @rafael-telles 

If there's no other comments, let's tentatively plan this for release on Wednesday? (Just to give time for any other review)

**Comment by lidavidm** [comment]
Retargeted for 2022/02/16 and added link from the 7.0.0 announcement.

**Comment by lidavidm** [comment]
CI will not pass until Wednesday because Jekyll isn't rendering future-dated posts.

**Review by kylepbit — APPROVED** [review]
This looks good to me, nice work @lidavidm.

**Comment by lidavidm** [comment]
And it's live: https://arrow.apache.org/blog/2022/02/16/introducing-arrow-flight-sql/

Thanks everyone!

**Pull request closed without merging**